### PR TITLE
Remove deprecated getMock in tests and generated code + new cs rule

### PR DESCRIPTION
**Pull request by antograssiot**

### CODE REVIEW

**Comment by antograssiot** [comment]
I can tag a new version once it will be merged to avoid to continue to generate deprecated code in tests.

**Comment by dereuromark** [comment]
Why hiding and not fixing the doc blocks?
Those are not void, but returning null etc as it looks like.

File: src/Shell/Task/ControllerTask.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ class ControllerTask extends BakeTask
      * Execution method always used for tasks
      *
      * @param string|null $name The name of the controller to bake.
-     * @return void
+     * @return void|bool
```
Comment (dereuromark) [inline src/Shell/Task/ControllerTask.php]: null|bool
void by definition cannot be mixed with others
either you are supposed to check on the return value (non-void) or you are not (void).
mixing void with non-void means this behavior is undefined ;)

File: src/Shell/Task/SimpleBakeTask.php [hunk 2]
```diff
@@ -111,7 +111,7 @@ public function bake($name)
      * Generate a test case.
      *
      * @param string $className The class to bake a test for.
-     * @return string|bool
+     * @return void|string|bool
```
Comment (dereuromark) [inline src/Shell/Task/SimpleBakeTask.php]: string|bool|null
I would always put the least expected to the end.

Comment (antograssiot) [inline src/Shell/Task/SimpleBakeTask.php]: changing void to null will unplease phpcs

File: composer.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
         "cakephp/cakephp": ">=3.2.0"
     },
     "require-dev": {
-        "cakephp/cakephp-codesniffer": "dev-master",
+        "cakephp/cakephp-codesniffer": "^2.1",
```
Comment (markstory) [inline composer.json]: Thanks :+1:

**Comment by markstory** [comment]
Looks good to me.

**Pull request merged**